client/src/components/AdvancedMultiSelect.js

```diff
@@ -0,0 +1,359 @@
+					label={fieldLabel}
+					component={renderInputField}
+					value={this.state.searchTerms}
+					placeholder={placeholder}
+					onChange={this.changeSearchTerms}
+					onFocus={this.handleInputFocus}
+					onBlur={this.handleInputBlur}
+					innerRef={el => {this.overlayTarget = el}}
+					extraColElem={renderExtraCol ? this.renderShortcutsTitle() : null}
+					addon={addon}
+				/>
+				<Overlay
+					show={this.state.showOverlay}
+					container={this.overlayContainer}
+					target={this.overlayTarget}
+					rootClose={true}
+					onHide={this.handleHideOverlay}
+					placement="bottom"
+					animation={false}
+					delayHide={200}
+				>
+					<Popover id={fieldName} title={null} placement="bottom" style={{width: '100%', maxWidth: '100%', boxShadow: '0 6px 20px hsla(0, 0%, 0%, 0.4)'}}>
+						<ContainerDimensions>
+							{({ width }) =>
+								<Row className="border-between">
+									<div className="sideBar">
+										<Col sm={4} md={3}>
+										{ width < MOBILE_WIDTH ?
+											<div>{console.log(width)}<SelectFilterInputField items={filterDefs} handleOnChange={this.handleOnChangeSelect}/></div>
+											:
+											<div>{console.log(width)}<FilterList items={filterDefs} currentFilter={this.state.filterType} handleOnClick={this.changeFilterType}/></div>
+										}
+										</Col>
+									</div>
+									<div className="multiSelectContent" style={{ minHeight: '80px' }}>
+										<Col sm={8} md={9}>
+											<this.props.overlayComponent
+												items={items}
+												selectedItems={selectedItems}
+												addItem={this.addItem}
+												removeItem={this.removeItem}
+												isLoading={isLoading}
+												loaderMessage={"No results found"}
+											/>
+											<footer className="searchPagination">
+												{this.paginationFor(filterType)}
+											</footer>
+										</Col>
+									</div>
+								</Row>
+							}
+						</ContainerDimensions>
+					</Popover>
+				</Overlay>
+				<AdvancedMultiSelectTarget overlayRef={el => {this.overlayContainer = el}}/>
+				<Row>
+					<Col sm={2} />
+					<Col sm={7}>{renderSelectedWithDelete}</Col>
+					<Col sm={3}>{renderExtraCol ? this.renderShortcuts() : null}</Col>
+				</Row>
+			</React.Fragment>
+		)
+	}
+
+	handleInputFocus = () => {
+		if (this.state.inputFocused === true) {
+			return
+		}
+		this.setState({
+			inputFocused: true,
+			showOverlay: true,
+			isLoading: true,
+		}, this.fetchResults())
+	}
+
+	handleInputBlur = () => {
+		this.setState({
+			inputFocused: false,
+		})
+	}
+
+	handleHideOverlay = () => {
+		if (this.state.inputFocused) {
+			return
+		}
+		this.setState({
+			filterType: Object.keys(this.props.filterDefs)[0],
+			searchTerms: '',
+			results: {},
+			isLoading: false,
+			showOverlay: false,
+		})
+	}
+
+	changeSearchTerms = (event) => {
+		// Reset the results state when the search terms change
+		this.setState({
+			isLoading: true,
+			searchTerms: event.target.value,
+			results: {}
```

Comment:
In commit c61a146 I removed resetting the results on purpose, as a result of the feedback from Vassil. See the commit message. But well, as we now have the loading indicator I no longer find it disturbing. What do you think @VassilIordanov ?

Reply:
I do not see any side effects of this so I will leave it as it is now.